app/uk/gov/hmrc/securemessage/controllers/SecureMessageController.scala

```diff
@@ -40,19 +45,19 @@ class SecureMessageController @Inject()(
 
   def createConversation(client: String, conversationId: String): Action[JsValue] = Action.async(parse.json) {
     implicit request =>
-      withJsonBody[ConversationRequest] { conversationRequest =>
-        secureMessageService.createConversation(conversationRequest.asConversation(client, conversationId)).map {
-          case Right(_)                        => Created
-          case Left(error: SecureMessageError) => handleErrors(client, conversationId, error)
-        }
-      }.recover {
-        case error: Exception => handleErrors(client, conversationId, error)
+      client match {
+        case "cdcm" =>
+          withJsonBody[CdcmConversation] { cdcmConversation =>
+            saveConversation(cdcmConversation.asConversation(client, conversationId))
+          }
+        case _ => Future(handleErrors(client, conversationId, InvalidRequestBody(s"Not supported client: $client")))
```

Comment:
Is this an invalid requestBody error? it looks more like an invalid query parameter value error, no?

Reply:
actually, not a query parameter at all, it's part of the URL path

Reply:
Could we have a test for this?